ui/ts/pages/helpus.ts

```diff
@@ -0,0 +1,215 @@
+              this.stage = Stage.ThanksSize;
+              // After 400ms display the "Thanks" modal text
+              setTimeout(
+                () => {
+                  this.stage = Stage.ThanksText;
+                  m.redraw();
+                },
+                400);
+            });
+          }
+        }
+
+        // If the close button is hit, or the escape button is pressed, close the modal and remove "help-us" from the URL.
+        close(): void {
+          this.userData.attributes.dismissed = true;
+          this.userData.save().then(() => {
+            // Remove the event listener that listens for the escape key.
+            document.removeEventListener("keydown", this.closeOnEscapeBound);
+            let routeParts: string[] = m.route().split("?");
+            let route: string = routeParts[0];
+            let params: Object = m.route.parseQueryString(routeParts[1]);
+            delete params["help-us"];
+            // Fade the correct modal.
+            if (this.stage < Stage.ModalClose) {
+              this.stage = Stage.ModalClose;
+            } else {
+              this.stage = Stage.ThanksClose;
+            }
+            // Remove help-us from the URL after the fade completes.
+            setTimeout(
+              () => {
+                m.route(route, params);
+              },
+              200);
+          });
+        }
+
+        // Runs the close function when the escape key is pressed.
+        closeOnEscape(e: KeyboardEvent): void {
+          if (e.keyCode === 27) {
+            this.close();
+            m.redraw();
+          }
+        }
+      }
+
+      export function controller(): ModalController {
+        return new ModalController();
+      }
+
+      export function view(ctrl: ModalController): _mithril.MithrilVirtualElement {
+        let modalClass: string = "";
+        let containerClass: string = "";
+
+        if (ctrl.stage === Stage.ThanksSize) {
+          modalClass += ".thanks-pre";
+        }
+        if (ctrl.stage === Stage.ModalStart) {
+          modalClass += ".hide-modal";
+        }
+        if (ctrl.stage === Stage.ModalClose || ctrl.stage === Stage.ThanksClose) {
+          containerClass += ".close";
+        }
+
+        let close: () => void = ctrl.close.bind(ctrl);
+
+        if (ctrl.show) {
+          // Banner
+          if (ctrl.stage === Stage.Banner) {
+            return m(".banner", [
+              m("span.checkmark", m.trust("&#x2713; ")),
+              "Check for updates and help us improve.",
+              m("button", {onclick: ctrl.clickBanner.bind(ctrl)}, "Opt In"),
+              m(".close", {onclick: close}, m.trust("&#x2715; ")),
+            ]);
+          // Help Us Form
+          } else if (ctrl.stage === Stage.Modal || ctrl.stage === Stage.ThanksSize || ctrl.stage === Stage.ModalStart || ctrl.stage === Stage.ModalClose) {
+            return m(".modal-container" + containerClass, [
```

Comment:
As mentioned in person, if you want to use the controller to bind and unbind the "closeOnEscape" function, the appropriate thing to do is to make the code in this section of the "if" statement (which occurs when a modal dialog is actually being displayed) into its own SubComponent.  That subComponent would be instantiated whenever the modal is displayed, and would be _unloaded_ when the modal is closed, making it appropriate for binding the "closeOnEscape".